tests/integration/testIntegration.nim

```diff
@@ -243,8 +256,53 @@ twonodessuite "Integration tests", debug1 = false, debug2 = false:
 
     let responsePast = client1.requestStorageRaw(cid, duration=1.u256, reward=2.u256, proofProbability=3.u256, collateral=200.u256, expiry=currentTime-10)
     check responsePast.status == "400 Bad Request"
-    check responsePast.body == "Expiry needs to be in future"
+    check "Expiry needs to be in future" in responsePast.body
 
     let responseBefore = client1.requestStorageRaw(cid, duration=1.u256, reward=2.u256, proofProbability=3.u256, collateral=200.u256, expiry=currentTime+10)
     check responseBefore.status == "400 Bad Request"
-    check responseBefore.body == "Expiry has to be before the request's end (now + duration)"
+    check "Expiry has to be before the request's end (now + duration)" in responseBefore.body
+
+  test "updating availability":
+    let availability = client1.postAvailability(totalSize=140000.u256, duration=200.u256, minPrice=300.u256, maxCollateral=300.u256).get
+
+    client1.patchAvailability(availability.id, duration=100.u256.some, minPrice=200.u256.some, maxCollateral=200.u256.some)
+
+    var updatedAvailability = (client1.getAvailabilities().get).findItem(availability).get
+    check updatedAvailability.duration == 100
+    check updatedAvailability.minPrice == 200
+    check updatedAvailability.maxCollateral == 200
+    check updatedAvailability.totalSize == 140000
+    check updatedAvailability.freeSize == 140000
+
+    # Non existing availability
```

Comment:
Instead of testing all these things in one `test`, I'd rather see them in their own `test`s. Now each of these is dependent on the state that is left behind by the others.

(would also be great if we could test this outside of the integration tests, but perhaps that's something for another day)

Reply:
I know, I was also thinking about it and I agree with you. The thing is the way how the `testIntegration` is structured - it is a "bundle of everything" 😅 So I didn't really want to pollute it with "many" test cases. Also, I was worried about the performance hit of having multiple small tests like this.

But okey, let me refactor it ;-)